libass/ass.c

```diff
@@ -641,6 +649,8 @@ static int process_style(ASS_Track *track, char *str)
         // probably an ancient script version
         if (track->track_type == TRACK_TYPE_SSA)
             track->style_format = strdup(ssa_style_format);
+        else if (track->track_type == TRACK_TYPE_ASS2)
+            track->style_format = strdup(ass2_style_format);
```

Comment:
If `style_format` was written via API, e.g. an API-created empty track with pre-filled format, this will still use whatever `track_type` even if the received data was v4++.

#651 avoids this by never checking `style_format` if we deal with v4++, ... although perhaps it is not necessary to defend against custom format lines in v4++ as strongly and this version is a bit simpler

(same for event format)